convert_hf_to_gguf.py

```diff
@@ -4899,7 +4911,7 @@ def set_vocab(self):
     def set_gguf_parameters(self):
         d_model = self.find_hparam(["hidden_size", "d_model", "dim"])
         d_conv  = self.find_hparam(["conv_kernel",       "d_conv"],  optional=True) or 4
-        d_inner = self.find_hparam(["intermediate_size", "d_inner"], optional=True) or 2 * d_model
+        d_inner = self.find_hparam(["mamba_d_ssm", "intermediate_size", "d_inner"], optional=True) or 2 * d_model
         d_state = self.find_hparam(["state_size",        "d_state"], optional=True) or 128
         head_dim = self.find_hparam(["head_dim"],                    optional=True) or 64
```

Comment:
I'm getting this error when trying to run a `Q8_0` conversion of <https://huggingface.co/tiiuae/Falcon-H1-1.5B-Instruct>.

(This does *not* happen in <https://huggingface.co/tiiuae/Falcon-H1-1.5B-Instruct-GGUF>, so it looks like a regression)

```
llama_model_load: error loading model: check_tensor_dims: tensor 'blk.0.ssm_in.weight' has wrong shape; expected  2048,  6680, got  2048,  6704,     1,     1
```

It looks like the problem is caused by a wrong time step rank being stored.

```suggestion
        head_dim = self.find_hparam(["mamba_d_head", "head_dim"],    optional=True) or 64
```

Reply:
Nice catch, thanks for fixing this. Somehow on the 0.5B which I was using to test, it was silently working